File: src/StripController/stripper/SetBrighnessCommand.cpp
#include "SetBrighnessCommand.h"

SetBrighnessCommand::SetBrighnessCommand(Strip* strip, Logger* logger)
	:_strip(strip), _logger(logger)
{
}

int SetBrighnessCommand::GetCommandId()
{
	return Command::SetBrighnessCommandId;
}

void SetBrighnessCommand::Handle(byte* command)
{
	_logger->Trace("Handle SetBrighnessCommand begin");

	int offset = 1;
	byte value = command[offset];

	char buffer[255];
	sprintf(buffer, "Arguments: {\"value\": %d}", value);
	_logger->Trace(buffer);

	_strip->SetBrightness(value);

	_logger->Trace("Handle SetBrighnessCommand end");
}

File: src/StripController/StripController.Infrastructure/StripWrapper/SetBrighnessCommand.cs
namespace StripController.Infrastructure.StripWrapper
{
    class SetBrighnessCommand : Command
    {
        private readonly byte _value;

        public SetBrighnessCommand(byte value)
        {
            _value = value;
        }

        public override byte[] GetBody()
        {
            return new[] { _value };
        }

        public override int GetCommandId()
        {
            return Commands.SetBrighnessCommandId;
        }
    }
}
